theano/gof/cc.py

```diff
@@ -1648,6 +1796,205 @@ def instantiate_code(self, n_args):
+        for var, name in zip(utils.uniq(self.outputs), args):
+            mapping_str += "// Output %(var)s->%(name)s\n" % locals()
+        if sys.platform != "win32":
+            for name in args:
+                out_print += """
+                //PyList_GET_ITEM return a borrowed reference
+                PyObject *tmp_%(name)s=PyList_GET_ITEM(struct_ptr->%(name)s, 0);
+                PyObject_Print(tmp_%(name)s, stdout, Py_PRINT_RAW);
+                printf("\\n");
+                """ % locals()
+        else:
+            # On Windows, when the python is compiled with a different run time
+            # (Visual Studio version?), then the executable, the stdout and
+            # stderr shouldn't not be shared between the 2 run time, so we
+            # can't call PyObject_Print(). Otherwise, it segfault.
+            for out, name in zip(self.outputs, args):
+                out_print += """
+   //PyList_GET_ITEM return a borrowed reference
+   PyObject *tmp_%(name)s=PyList_GET_ITEM(struct_ptr->%(name)s, 0);
+   PyObject *str_%(name)s = PyObject_Str(tmp_%(name)s);
+   //PyString_AsString return a ptr to the internal representation.
+   printf("%%s\\n", PyString_AsString(str_%(name)s));
+   Py_CLEAR(str_%(name)s);
+                """ % locals()
+        main = """
+%(mapping_str)s
+int main(int argc, char *argv[]) {
+ Py_SetProgramName(argv[0]);  /* optional but recommended */
+ Py_Initialize();
+
+ // Those print are there to help debug import of python module
+ PyObject * numpy = PyImport_ImportModule("numpy");
+ printf("After import numpy %%p\\n", numpy);
+ PyErr_Print();
+ //import_array{,1,2} can be called many times without problems.
+ import_array1(1);
+ printf("after import_array1()\\n");
+
+ %(struct_name)s *struct_ptr = cinit();
+ int run_ret = 0;
+ if(struct_ptr){
+   %(shared_load)s
+   %(in_init)s
+   printf("after cinit()\\n");
+   //Function execution
+   run_ret = struct_ptr->run();
+   printf("run() from the shared library returned=%%d\\n", run_ret);
+
+   if(run_ret==0){
+   %(out_print)s
+   }else if(run_ret != 0){
+     // See out_print to know why we can't call PyObject_Print on win32
+     PyObject *str_err = PyObject_Str(struct_ptr->__ERROR);
+     //PyString_AsString return a ptr to the internal representation.
+     printf("Error: %%s\\n", PyString_AsString(str_err));
+     Py_CLEAR(str_err);
+   }
+ }else{
+   printf("cinit() failed!\\n");
+   return 1;
+ }
+ delete struct_ptr;
+
+ printf("main end, before Py_Finalize\\n");
+ Py_Finalize();
+ return run_ret;
+}
+        """ % dict(struct_name=self.struct_name,
+                   **locals())
+        # TODO, should struct_ptr.cleanup() cleanup the __ERROR structure
+        return main
+
+    def cinit_code(self):
+        code = StringIO()
+        n_args = len(self.args)
+        struct_name = self.struct_name
+        param = ','.join('PyObject * arg_%i' % n for n in range(n_args)), ');'
+        symbol2r = {v: k for (k, v) in self.r2symbol.items()}
```

Comment:
This isn't valid python 2 syntax.

I think this would work:

`dict([(v, k) for (k, v) in self.r2symbol.items()])`